python/ray/data/_internal/logical/operators/map_operator.py

```diff
@@ -0,0 +1,113 @@
+import sys
+from typing import Any, Dict, Iterable, Iterator, Optional, Union
+
+import ray
+from ray.data._internal.compute import BlockTransform
+from ray.data._internal.execution.interfaces import PhysicalOperator
+from ray.data._internal.execution.operators.map_operator import MapOperator
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.data._internal.compute import (
+    get_compute,
+    CallableClass,
+    ComputeStrategy,
+    TaskPoolStrategy,
+    ActorPoolStrategy,
+)
+from ray.data.block import BatchUDF, Block
+
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+class MapBatches(LogicalOperator):
+    """Logical operator for map_batches."""
+
+    def __init__(
+        self,
+        block_fn: BlockTransform,
+        fn: BatchUDF,
+        batch_size: Optional[Union[int, Literal["default"]]] = "default",
+        compute: Optional[Union[str, ComputeStrategy]] = None,
+        batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
+        zero_copy_batch: bool = False,
+        target_block_size: Optional[int] = None,
+        fn_args: Optional[Iterable[Any]] = None,
+        fn_kwargs: Optional[Dict[str, Any]] = None,
+        fn_constructor_args: Optional[Iterable[Any]] = None,
+        fn_constructor_kwargs: Optional[Dict[str, Any]] = None,
+        ray_remote_args: Optional[Dict[str, Any]] = None,
+    ):
+        super().__init__("MapBatches", [])
+        self._block_fn = block_fn
+        self._fn = fn
+        self._batch_size = batch_size
+        self._compute = compute or "tasks"
+        self._batch_format = batch_format
+        self._zero_copy_batch = zero_copy_batch
+        self._target_block_size = target_block_size
+        self._fn_args = fn_args
+        self._fn_kwargs = fn_kwargs
+        self._fn_constructor_args = fn_constructor_args
+        self._fn_constructor_kwargs = fn_constructor_kwargs
+        self._ray_remote_args = ray_remote_args or {}
+
+    def get_physical_dag(self) -> PhysicalOperator:
```

Comment:
This is largely copied from `legacy_compat.py:_stage_to_operator()` for now. Let me know if we want to create utility method instead of copying.